code/modules/surgery/organs/blood.dm

```diff
@@ -70,13 +70,21 @@
 			if(BP.internal_bleeding)
 				internal_bleeding_rate += 0.5
 
-		bleed_rate = max(bleed_rate - 0.5, temp_bleed)//if no wounds, other bleed effects (heparin) naturally decreases
+		bleed_rate = max(bleed_rate - 0.5, temp_bleed * get_bleed_modifier()) //if no wounds, other bleed effects (heparin) naturally decreases
```

Comment:
`bleed_rate` is a `/mob/living/carbon/human` field, not a local variable.
So if you get calzeo in your blood for a single tick, `bleed_rate` would be immediately set to zero (`max(1234 - 0.5, 1234 * 0) == 0`), and stay there even after the chemical leaves your body.
I would suggest something like
```dm
bleed_rate = max(bleed_rate - 0.5, temp_bleed)  // bleeding heals naturally
temp_bleed = bleed_rate * get_bleed_modifier()  // bleed modifier is temporary and does accumulate across ticks
internal_bleeding_rate *= get_bleed_modifier()
//..
bleed(temp_bleed)
```

Reply:
I made sure to test it before pushing, it resets after the chemical leaves. I don't see why this change would be necessary?